assets/src/edit-story/components/panels/pageAttachment/index.js

```diff
@@ -0,0 +1,192 @@
+  const {
+    currentPage,
+    updateCurrentPageProperties,
+    updateElementsById,
+  } = useStory((state) => ({
+    updateCurrentPageProperties: state.actions.updateCurrentPageProperties,
+    updateElementsById: state.actions.updateElementsById,
+    currentPage: state.state.currentPage,
+  }));
+  const { setDisplayLinkGuidelines } = useCanvas((state) => ({
+    setDisplayLinkGuidelines: state.actions.setDisplayLinkGuidelines,
+  }));
+  const { pageAttachment = {} } = currentPage;
+  const defaultCTA = __('Learn more', 'web-stories');
+  const { url, ctaText = defaultCTA } = pageAttachment;
+  const [_ctaText, _setCtaText] = useState(ctaText);
+  const [displayWarning, setDisplayWarning] = useState(false);
+
+  const { getElementsInAttachmentArea } = useElementsWithLinks();
+  const linksInAttachmentArea = getElementsInAttachmentArea();
+  const hasLinksInAttachmentArea = linksInAttachmentArea.length > 0;
+
+  useEffect(() => {
+    return () => {
+      setDisplayLinkGuidelines(false);
+    };
+  }, [hasLinksInAttachmentArea, setDisplayLinkGuidelines, url]);
+
+  const onFocus = () => {
+    if (hasLinksInAttachmentArea && !url?.length) {
+      setDisplayWarning(true);
+      setDisplayLinkGuidelines(true);
+    }
+  };
+
+  useEffect(() => {
+    if (displayWarning && url?.length) {
+      setDisplayWarning(false);
+    }
+  }, [url, displayWarning]);
+
+  const updatePageAttachment = useCallback(
+    (value) => {
+      const updatedValue = value;
+      if (value.url) {
+        const urlWithProtocol = withProtocol(value.url);
+        const valid = isValidUrl(urlWithProtocol);
+        updatedValue.url = urlWithProtocol;
+        setIsInvalidUrl(!valid);
+        if (hasLinksInAttachmentArea) {
+          // Remove links from elements if Page attachment was updated.
+          updateElementsById({
+            elementIds: linksInAttachmentArea.map(({ id }) => id),
+            properties: {
+              link: null,
+            },
+          });
+        }
+      }
+      const _pageAttachment = {
+        ...pageAttachment,
+        ...updatedValue,
+      };
+      updateCurrentPageProperties({
+        properties: { pageAttachment: _pageAttachment },
+      });
+    },
+    [
+      updateCurrentPageProperties,
+      pageAttachment,
+      hasLinksInAttachmentArea,
+      linksInAttachmentArea,
+      updateElementsById,
+    ]
+  );
+
+  const [isInvalidUrl, setIsInvalidUrl] = useState(
+    !isValidUrl(withProtocol(url || ''))
+  );
+
+  const isDefault = _ctaText === defaultCTA;
+  return (
+    <SimplePanel
+      name="pageAttachment"
+      title={__('Page Attachment', 'web-stories')}
+    >
+      <Row>
+        <Note>
+          {__('Type an address to add a page attachment', 'web-stories')}
+        </Note>
+      </Row>
+
+      <Row>
+        <ExpandedTextInput
+          placeholder={__('Web address', 'web-stories')}
+          onChange={(value) => updatePageAttachment({ url: value })}
```

Comment:
Note to self: it probably makes sense to create a Link input instead once #2745 gets fixed.